src/order/compactly_generated.lean

```diff
@@ -470,6 +473,21 @@ theorem is_complemented_of_Sup_atoms_eq_top (h : Sup {a : α | is_atom a} = ⊤)
         refine hc2.directed_on.mono (λ s t, Sup_le_Sup) } },
     { rcases set.mem_sUnion.1 ha with ⟨s, sc, as⟩,
       exact (hc1 sc).2.2 a as } }
+end
+
+lemma exists_set_independent_of_Sup_atoms_eq_top (h : Sup {a : α | is_atom a} = ⊤) :
+  ∃ s : set α, complete_lattice.set_independent s ∧ Sup s = ⊤ ∧ ∀ a ∈ s, is_atom a :=
+begin
+  obtain ⟨s, s_ind, s_top, s_atoms⟩ := exists_set_independent_is_compl_Sup_atoms h (⊥ : α),
+  refine ⟨s, s_ind, eq_top_of_is_compl_bot (is_compl.symm s_top), s_atoms⟩,
+end
+
+/-- See Theorem 6.6, Călugăreanu -/
+lemma is_complemented_of_Sup_atoms_eq_top
+  (h : Sup {a : α | is_atom a} = ⊤) : is_complemented α :=
+⟨λ b, begin
+  obtain ⟨s, s_ind, s_top, s_atoms⟩ := exists_set_independent_is_compl_Sup_atoms h b,
+  refine ⟨Sup s, s_top⟩
```

Comment:
Couldn't these just be `exact`, since they end the proof?
```suggestion
  exact ⟨s, s_ind, eq_top_of_is_compl_bot (is_compl.symm s_top), s_atoms⟩
end

/-- See Theorem 6.6, Călugăreanu -/
lemma is_complemented_of_Sup_atoms_eq_top
  (h : Sup {a : α | is_atom a} = ⊤) : is_complemented α :=
⟨λ b, begin
  obtain ⟨s, s_ind, s_top, s_atoms⟩ := exists_set_independent_is_compl_Sup_atoms h b,
  exact ⟨Sup s, s_top⟩
```

Reply:
BTW, we can use `let ⟨s, s_ind, s_top, s_atoms⟩ := ... in ...` instead of `begin/obtain/exact`.